Added Gujarati translation

Darpana-iitgn opened the pull request:
Added Gujarati translation as part of issue #279

imblowfish commented:
Hello @Darpana-iitgn could you please revert changes in `package.json` and `package-lock.json`?

imblowfish commented on src/utils/translations/gu.ts:
```diff
@@ -0,0 +1,94 @@
+  Period: "માસિક",
+  today: "આજે",
+  Delay: "વિલંબ",
+  Days: "(૦)[દિવસ];(૧)[દિવસ];(૨-૨)[દિવસ];(૫-૨૦)[દિવસ];(૨૧)[દિવસ];(૨૨-૨૪)[દિવસ];(૨૫-૩૦)[દિવસ];(૩૧)[દિવસ];(૩૨-૩૪)[દિવસ];(૩૫-૪૦)[દિવસ];",
```
You are writing the same word in any case, so you just can use `(0-inf)` instead

IraSoro commented:
Hello, @Darpana-iitgn! I looked at your translation and it seems to me that it is not quite correct. Unfortunately, I don't know Gujarati, so I used the ChatGPT to check, and it suggested a slightly different version.

For example, ChatGPT translated the phrase "Period in 10 days" as "10 દિવસમાં માસિક આવશે", and in your version "માસિક ધારામાં 10 દિવસ" it seems to have a slightly different meaning - more about period that last 10 days, and not that it will start in 10 days. Here is screenshot that the chat showed:
![{2F25A257-D4A8-4705-A96E-E869C18F10C4}](https://github.com/user-attachments/assets/9eeddcd9-f9b6-4da4-a772-e851489024a6)
There are also other comments on the translation.

Maybe the chat is wrong? Could you carefully check the translation again?

Thanks!

IraSoro reviewed (changes requested):
PR didn't pass the formatter check. If it's not difficult, could you fix this, please? To do this, you can run
```
npm run fmt
```

IraSoro commented:
Hello, @Darpana-iitgn! Thank you for your interest in the project! 

I looked at your translation and it seems to me that it is not quite correct. Unfortunately, I don't know Gujarati, so I used the ChatGPT to check, and it suggested a slightly different version.

For example, ChatGPT translated the phrase "Period in 10 days" as "10 દિવસમાં માસિક આવશે", and in your version "માસિક ધારામાં 10 દિવસ" it seems to have a slightly different meaning - more about period that last 10 days, and not that it will start in 10 days. Here is screenshot that the chat showed:
![{2F25A257-D4A8-4705-A96E-E869C18F10C4}](https://github.com/user-attachments/assets/9eeddcd9-f9b6-4da4-a772-e851489024a6)
There are also other comments on the translation.

Maybe the chat is wrong? Could you carefully check the translation again?

Thanks!

Darpana-iitgn commented:
Hi @IraSoro, I had misinterpreted the code earlier and have now corrected this line. The suggestion by ChatGPT is correct, but we can also translate "Period in 10 days" as "માસિક ધારા 10 દિવસમાં", which is what I have changed the code to. I have checked other translations once again and they seem correct to me, but pls let me know if I have missed anything else. I ran "npm run fmt", but it is causing changes in various files so I didnt commit it.

IraSoro commented:
@Darpana-iitgn, I checked again and found the following errors:
![{99B7180B-342F-4BB5-8867-268E89EF7A9D}](https://github.com/user-attachments/assets/e894fa72-7cb1-47fd-b99e-037d99fdb6eb)

![{2C298049-BBE7-4C1D-A203-386AD143375B}](https://github.com/user-attachments/assets/dc6352e9-557c-4f2d-b457-d5c7aa81a6c1)

![image](https://github.com/user-attachments/assets/04ec1272-7c48-4a4c-8a2f-815ef7c62488)

I have a question, are you a native speaker? Because for me, as a non-native speaker, the translation doesn't look quite correct. It loses the context of the topic, mistakes in the description of menstruation and other errors caused by either typos or automatic translation.

Darpana-iitgn commented:
Hi, I am a native speaker and have done all the translations myself. ChatGPT is not an extremely reliable source to check since it gives varying suggestions to all users, which might not necessarily be always correct. In the first 2 pictures you have shown, native speakers would consider both the sentences be the same in meaning, since sometimes similar sentences can have the same translation in english. 3rd error was a spelling mistake. 

Still, I have changed my code to the version ChatGPT recommends and have checked it too.
![image](https://github.com/user-attachments/assets/016821a7-b1f4-49d2-872f-0a4d6945a146)

IraSoro commented:
@Darpana-iitgn, thanks for replying to the comment and correcting the errors!

Congratulations on your first contribution to `Peri`! And thanks for translating to Gujarati!
